packages/react-ui-core/src/Gmap/FreeDrawLayer.js

```diff
@@ -0,0 +1,202 @@
+    map.setOptions({
+      draggable: true,
+      zoomControl: true,
+      scrollwheel: true,
+      disableDoubleClickZoom: true,
+      ...mapControls,
+    })
+    this.setState({
+      drawing: false,
+    })
+  }
+
+  disableMapControls() {
+    const { map } = this.props
+    map.setOptions({
+      draggable: false,
+      zoomControl: false,
+      scrollwheel: false,
+      disableDoubleClickZoom: false,
+    })
+    this.setState({
+      drawing: true,
+    })
+  }
+
+  @autobind
+  enableMapDraw() {
+    const { map, multipleShapes } = this.props
+    this.disableMapControls()
+
+    if (!multipleShapes) {
+      this.clearAllShapes()
+    }
+    setupEvents(map, EVENTS, {
+      onMouseDown: () => {
+        this.drawFreeHand()
+      },
+    }, true)
+  }
+
+  drawFreeHand() {
+    const { map, onMapDrawStart, dataStyle, onMapDrawEnd } = this.props
+    const poly = new window.google.maps.Polyline({
+      map,
+      clickable: false,
+      ...dataStyle.polyline,
+    })
+    const drawPoly = new window.google.maps.Polygon({
+      map,
+      path: poly.getPath(),
+      clickable: false,
+      ...dataStyle.polylineFill,
+    })
+
+    if (onMapDrawStart) {
+      onMapDrawStart()
+    }
+
+    const listener = setupEvents(map, EVENTS, {
+      onMouseMove: e => {
+        poly.getPath().push(e.latLng)
+      },
+      onTouchStart: e => {
+        poly.getPath().push(e.latLng)
+      },
+    }, false)
+
+    const onMouseUp = () => {
+      removeEvent(listener.onMouseMove)
+      poly.setMap(null)
+      drawPoly.setMap(null)
+      if (onMapDrawEnd) {
+        onMapDrawEnd(this.getPolygonCoords(poly))
+      }
+      this.enableMapControls()
+    }
+
+    setupEvents(map, EVENTS, {
+      onMouseUp,
+      onTouchEnd: onMouseUp,
+    }, true)
+  }
+
+  addShapeToGMaps(path) {
+    const { map, dataStyle } = this.props
+    return new window.google.maps.Polygon({
+      map,
+      path,
+      ...dataStyle.polygon,
+    })
+  }
+
+  addShape(path) {
+    const rawPolygon = this.addShapeToGMaps(path)
+    this.polygons = [...this.polygons, rawPolygon]
+  }
+
+  @autobind
+  clearAllShapes() {
+    for (let i = 0; i < this.polygons.length; i += 1) {
+      this.polygons[i].setMap(null)
+    }
+
+    this.polygons = []
+  }
+
+  formatLongLatToGmapsCoordinates(shapes) {
+    // Should be refactored later to
+    // over(pipe(allArr, allArr), point => ({lat: point[1], lng: point[0]})(Object.values(shape)
+
+    return Object.values(shapes)
+      .reduce((polygonAcc, polygon) => {
+        polygonAcc.push(polygon.reduce((pointAcc, point) => {
+          pointAcc.push({ lat: point[1], lng: point[0] })
+          return pointAcc
+        }, []))
+        return polygonAcc
+      }, [])
+  }
+
+  render() {
+    const { theme, disabled } = this.props
+    const { drawing } = this.state
+    // This button is temporary - will be replaced with the tool tip when built
+    return (
+      <button
```

Comment:
Can we use the react-ui-core Button here?

Reply:
That will be happening in another pr @bethstevic worked on https://github.com/rentpath/react-ui/pull/337